be/src/exprs/subfield_expr.cpp

```diff
@@ -39,9 +39,23 @@ class SubfieldExpr final : public Expr {
 
         ASSIGN_OR_RETURN(ColumnPtr col, _children.at(0)->evaluate_checked(context, chunk));
 
-        // Enter multiple subfield for struct type, remain last subfield
-        for (size_t i = 0; i < _used_subfield_names.size() - 1; i++) {
-            std::string fieldname = _used_subfield_names[i];
+        // handle nullable column
+        std::vector<uint8_t> null_flags;
+        size_t num_rows = col->size();
+        null_flags.resize(num_rows, false);
+
+        for (size_t i = 0; i < _used_subfield_names.size(); i++) {
+            const std::string& fieldname = _used_subfield_names[i];
+
+            // merge null flags for each level
+            if (col->is_nullable()) {
+                auto* nullable = down_cast<NullableColumn*>(col.get());
```

Comment:
Why not use FunctionHelper::union_nullable_column or FunctionHelper::union_null_column?

Reply:
fixed